src/components/UserForm/UserForm.js

```diff
@@ -31,92 +36,141 @@ class UserForm extends React.Component {
     emailHandler = (e) => {
         this.setState({ email: e.target.value });
     };
-    
-    // photoUrlHandler = (e) => {
-    //     this.setState({ photoUrl: e.target.value });
-    // };
+
+    photoHandler = (e) => {
+        this.setState({ photo: e.target.value });
+    };
 
     clickHandler = (e) => {
         e.preventDefault();
-        if (!this.state.phone || !this.state.name || !this.state.surname || !this.state.email) {
+        if (!this.state.phone || !this.state.name || !this.state.surname || !this.state.email || !this.state.photo) {
             alert('Fill in all the fields');
             return;
         }
-       
+        if(!EmailValidator.validate(this.state.email)){
+            alert('invalid email');
+            this.setState({email: ''});
+            return;
+        }
+        if(this.state.phone.includes(' ')){
+           alert('ivalid number');
+           return;
+        }
+
         this.setState({
             name: '',
             surname: '',
             phone: '',
             email: '',
-            // photoUrl: '',
+            photo: '',
             users: this.state.users.concat([
                 {
                     name: this.state.name,
                     surname: this.state.surname,
                     phone: this.state.phone,
                     email: this.state.email,
-                    // photoUrl: this.state.photoUrl,
+                    isOnline: this.state.isOnline,
+                    photo: this.state.photo
                 },
             ]),
         });
     };
 
+    deleteHandler = (index) => {
+        let us = [...this.state.users];
+        us.splice(index, 1);
+        this.setState({ users: us });
+    };
+
+    popupHandler = (index) => {
+        this.setState({
+            edit: !this.state.edit,
+            editData: this.state.users[index],
```

Comment:
you don't need to keep editData variable in your state. just use this.state.users and index